src/main/java/com/puppycrawl/tools/checkstyle/JavadocDetailNodeParser.java

```diff
@@ -486,11 +498,12 @@ private static Token getMissedHtmlTag(RecognitionException exception) {
         final Deque<Token> stack = new ArrayDeque<>();
         for (int i = 0; i < tokenList.size(); i++) {
             final Token token = tokenList.get(i);
-            if (token.getType() == JavadocTokenTypes.HTML_TAG_NAME
+            //JavadocTokenTypes.HTML_TAG_NAME != JavadocParser.HTML_TAG_NAME. See Issue #5139.
+            if (token.getType() == JavadocParser.HTML_TAG_NAME
```

Comment:
Same here.

Also why is this using parser and not the token types? Is this method called before `getTokenType` and thats why?
Won't this get confusing as we need to swap tokens in 2 places instead of 1?
Is there anyway we can make it swap only in one place and assume the token types are correct everywhere else?

Reply:
Actually this is because `token` is the token object form `ANTLR`. This method is called for error reporting and there's no `DetailNode` tree at hand. An `ANTLR` token would continue to have the token value generated by ANTLR. We wouldn't have to perform updates to this line with addition of tokens for HTML tags in grammar because `token.getType()` and `JavadocParser.HTML_TAG_NAME` are both ANTLR units and will be automatically taken care of.

Reply:
Rather it was like since `JavadocTokenTypes.HTML_TAG_NAME` == `JavadocParser.HTML_TAG_NAME` we were previously able to use `JavadocTokenTypes.HTML_TAG_NAME`, but now since those two values have changed, we are forced to use value directly from the `JavadocParser`.